### PR TITLE
Add --no-headers to list subcommand for taskruns and pipelineruns

### PR DESCRIPTION
**Pull request by waveywaves**
# Changes

partly fixes https://github.com/tektoncd/cli/issues/799 

Add `--no-headers` flag to `list` subcommand for taskruns and pipelineruns.

##### Note
In the future, will open PRs related to https://github.com/tektoncd/cli/issues/799 https://github.com/tektoncd/cli/issues/785 together. i.e. for PR related to task, pipeline will have changes wrt `--all-namespaces` and `--no-headers`

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
Adds --no-headers flag for list subcommand under taskrun and pipelinerun
```

### CODE REVIEW

File: pkg/cmd/pipelinerun/list.go [hunk 1]
```diff
@@ -33,30 +33,29 @@ import (
...
 )
 
-const listTemplate = `{{- $prl := len .PipelineRuns.Items }}{{ if eq $prl 0 -}}
+const listTemplate = `{{- $prl := len .PipelineRuns.Items -}}{{- if eq $prl 0 -}}
 No PipelineRuns found
 {{- else -}}
-{{- if $.AllNamespaces }}NAMESPACE	NAME	STARTED	DURATION	STATUS	
-{{- else -}}
+{{- if not $.NoHeaders -}}
+{{- if $.AllNamespaces -}}
```
Comment (chmouel) [inline pkg/cmd/pipelinerun/list.go]: those templates are getting more complex than go codes 🤔 

(it's not an issue with your PR just a thought)

File: pkg/cmd/pipelinerun/list_test.go [hunk 2]
```diff
@@ -235,7 +247,7 @@ func TestListPipeline_empty(t *testing.T) {
 		t.Errorf("Unexpected error: %v", err)
 	}
 
-	test.AssertOutput(t, "No PipelineRuns found\n", output)
```
Comment (chmouel) [inline pkg/cmd/pipelinerun/list_test.go]: this should have a `\n` when printing, was somthing removed?

File: pkg/cmd/taskrun/testdata/TestListTaskRuns-empty_list.golden [hunk 3]
```diff
@@ -1 +1 @@
-No TaskRuns found
+No TaskRuns found
```
Comment (chmouel) [inline pkg/cmd/taskrun/testdata/TestListTaskRuns-empty_list.golden]: Same as before, no taskruns found, should have a carriage return.. can you double check please?

File: pkg/cmd/pipelinerun/list.go (hunk 1 above)
Comment (waveywaves) [reply pkg/cmd/pipelinerun/list.go]: I introduced templates in the list command for these two (or one of them can't remmeber exactly) for the --all-namespaces flag, it has made formatting better but in the PR I have refactored the templates to be cleaner and make more sense.

File: pkg/cmd/pipelinerun/list_test.go (hunk 2 above)
Comment (waveywaves) [reply pkg/cmd/pipelinerun/list_test.go]: The formatting in the go templatewas changed, updated with correct test and formatting.

**Comment by waveywaves** [comment]
@chmouel let me know as this is ready to be approved if there are not any changes to be made. I would like to split this commit into multiple commits considering the number of changes.

File: pkg/cmd/pipelinerun/list.go [hunk 4]
```diff
@@ -146,7 +145,7 @@ List all PipelineRuns in a namespace 'foo':
 	c.Flags().StringVarP(&opts.LabelSelector, "label", "", opts.LabelSelector, "A selector (label query) to filter on, supports '=', '==', and '!='")
 	c.Flags().BoolVarP(&opts.Reverse, "reverse", "", opts.Reverse, "list pipelineruns in reverse order")
 	c.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", opts.AllNamespaces, "list pipelineruns from all namespaces")
-
+	c.Flags().BoolVarP(&opts.NoHeaders, "no-headers", "", opts.NoHeaders, "when using default or custom formatting, don't print headers (default print headers)")
```
Comment (danielhelfand) [inline pkg/cmd/pipelinerun/list.go]: Maybe change the description to:

```suggestion
	c.Flags().BoolVarP(&opts.NoHeaders, "no-headers", "", opts.NoHeaders, "do not print column headers with output (default print column headers with output)")
```

File: pkg/cmd/taskrun/list.go [hunk 5]
```diff
@@ -145,6 +147,7 @@ List all TaskRuns of Task 'foo' in namespace 'bar':
 	c.Flags().StringVarP(&opts.LabelSelector, "label", "", opts.LabelSelector, "A selector (label query) to filter on, supports '=', '==', and '!='")
 	c.Flags().BoolVarP(&opts.Reverse, "reverse", "", opts.Reverse, "list taskruns in reverse order")
 	c.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", opts.AllNamespaces, "list taskruns from all namespaces")
+	c.Flags().BoolVarP(&opts.NoHeaders, "no-headers", "", opts.NoHeaders, "when using default or custom formatting, don't print headers (default print headers)")
```
Comment (danielhelfand) [inline pkg/cmd/taskrun/list.go]: Maybe change the description to:

```suggestion
	c.Flags().BoolVarP(&opts.NoHeaders, "no-headers", "", opts.NoHeaders, "do not print column headers with output (default print column headers with output)")
```

File: pkg/cmd/pipelinerun/list.go (hunk 4 above)
Comment (waveywaves) [reply pkg/cmd/pipelinerun/list.go]: Done.

File: pkg/cmd/taskrun/list.go (hunk 5 above)
Comment (waveywaves) [reply pkg/cmd/taskrun/list.go]: Done.

**Pull request merged**